pyteal/ast/abi/bool.py

```diff
@@ -0,0 +1,98 @@
+from typing import Union, cast, Sequence
+
+from ...types import TealType
+from ..expr import Expr
+from ..int import Int
+from ..bytes import Bytes
+from ..binaryexpr import GetBit
+from ..ternaryexpr import SetBit
+from .type import Type
+from .uint import NUM_BITS_IN_BYTE
+
+
+class Bool(Type):
+    def __init__(self) -> None:
+        super().__init__(TealType.uint64)
+
+    def new_instance(self) -> "Bool":
+        return Bool()
+
+    def has_same_type_as(self, other: Type) -> bool:
+        return type(other) is Bool
+
+    def is_dynamic(self) -> bool:
+        return False
+
+    def byte_length_static(self) -> int:
+        # Not completely accurate since up to 8 consecutive bools will fit into a single byte
+        return 1
+
+    def get(self) -> Expr:
+        return self.stored_value.load()
+
+    def set(self, value: Union[bool, Expr, "Bool"]) -> Expr:
+        if type(value) is bool:
+            value = Int(1 if value else 0)
+        if type(value) is Bool:
+            value = value.get()
+        return self.stored_value.store(cast(Expr, value))
+
+    def decode(
+        self,
+        encoded: Expr,
+        *,
+        startIndex: Expr = None,
+        endIndex: Expr = None,
+        length: Expr = None
+    ) -> Expr:
+        if startIndex is None:
+            startIndex = Int(0)
+        return self.decodeBit(encoded, startIndex * Int(NUM_BITS_IN_BYTE))
+
+    def decodeBit(self, encoded, bit: Expr) -> Expr:
+        return self.set(GetBit(encoded, bit))
+
+    def encode(self) -> Expr:
+        return SetBit(Bytes(b"\x00"), Int(0), self.get())
+
+    def __str__(self) -> str:
+        return "bool"
+
+
+def boolAwareStaticByteLength(types: Sequence[Type]) -> int:
+    length = 0
+    ignoreNext = 0
+    for i, t in enumerate(types):
+        if ignoreNext > 0:
+            ignoreNext -= 1
+            continue
+        if type(t) is Bool:
+            numBools = consecutiveBools(types, i)
+            ignoreNext = numBools - 1
+            length += boolSequenceLength(numBools)
+            continue
+        length += t.byte_length_static()
+    return length
```

Comment:
nit: personally prefer to use `for i in range(...)` and just let `i += numBools - 1` to skip iteration. Can ignore this, not a big deal.

Reply:
I don't think that actually skips iteration? I believe it only changes the current value of `i` and does not advance the iterator.

Reply:
oof... I was thinking I am writing cpp...